### ISSUE TITLE
Query throws UnknownError Exception

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13086]
When running the following query against an empty database:

```cypher
WITH 0 AS n1 ORDER BY n1
MATCH () WHERE false
WITH
    CASE 0
        WHEN n1 THEN 0
    END AS n3
    ORDER BY n3
RETURN 0
```

I encountered the following error:
```
Expected a sorted plan but got
.apply(fromSubquery = false)
.|.argument()
.projection(project = Seq("NULL AS `  UNNAMED0`"), discard = Set("n1"))
.limit(0)
.apply(fromSubquery = false)
.|.argument()
.sort(Seq(Ascending("n1")))
.projection("$`  AUTOINT0` AS n1")
.argument()
.build()
```

As the aforementioned query is syntactically and semantically correct I believe that no error should be thrown here.
I encountered this issue when testing queries against the **Neo4j 5.5.0 community version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
WITH 0 AS n1 ORDER BY n1
MATCH () WHERE false
WITH
    CASE 0
        WHEN n1 THEN 0
    END AS n3
    ORDER BY n3
RETURN 0
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `Neo4jError: Neo.DatabaseError.General.UnknownError (Expected a sorted plan but got...`

**Comment by smietana** [comment]
Thank you! We will investigate this.

**Comment by Dunfluff** [comment]
The query produces a `WHERE False` predicate which affected the planning of `Sort` and produced the error message, this has now been fixed in 5.8 and will instead finish planning and return 0 results. Thank you for finding it :)
